params/config.go

```diff
@@ -269,7 +269,6 @@ var (
 		CancunTime:                    nil,
 		PragueTime:                    nil,
 		VerkleTime:                    nil,
-		Cel2Time:                      newUint64(0),
```

Comment:
Please revert this, we want to enable `Cel2` for as many tests as possible.

If a test fails please opt-out by using `TestChainConfigNoCel2`.

Reply:
Ok I've raised a PR [here](https://github.com/celo-org/op-geth/pull/147) that reverts this change and makes use of `TestChainConfigNoCel2`.  It's not as much changes as I was envisaging but it's still quite a lot of extra scattered changes which will make updating from upstream more difficult. 

If the approach to fixing failed tests is simply to rely on `TestChainConfigNoCel2` then eventually (when all tests are using `TestChainConfigNoCel2`) we'll end up in the same situation as just not setting Cel2Time in the TestChainConfig but with a larger diff.

Also the approach of using `TestChainConfigNoCel2` is not particularly precise because sometimes we are setting it in a utility function and so it's effect is not confined to just the failing test.

So I think we would save ourselves some trouble if we didn't  add `Cel2Time` to the `TestConfig` and instead be aware of our specific changes and ensure that we test around them appropriately.

What do you think?

Just for reference, current changes we have that rely on `Cel2Time` are:

* Not setting gasLimit and difficulty to defaults in `genesis.ToBlock`
* Selecting transaction signers
* Setting exchange rates field in the block context
* Checking that the fee currency is whitelisted during the state transition
* Selecting the fee handler as the recipient of the base fee
* Adding celo precompiles to the set of precompiles acessible by the vm

Reply:
I still think it's worth having Cel2 flag enabled by default, so that we find potential errors when merging upstream.

I'd go with those changes that you did in the other branch. If we run into lot's of problems in the future during rebases, we can still change this.

Reply:
Ok will merge that other branch in here.

Reply:
Ok so actually this turned out to be a much larger task than it originally looked. Locally the go test caching failed me showing that tests were passing for https://github.com/celo-org/op-geth/pull/147, but actually they were not.

One of the problems with this is that the code panics which stops execution of the tests in a package, so you can't easily find all the failing tests, you have to fix one and then try again. Also now the diff is 4 times larger than #147 and tests are still failing, its not clear how much more will need to be changed.

So my feeling is that setting TestChainConfigNoCel2 on an individual test basis is not a sustainable approach.

@carterqw2 do you have any thoughts?